connectors/src/lib/data_sources.ts

```diff
@@ -229,43 +232,101 @@ async function _updateDocumentParentsField({
   }
 }
 
-const MAX_SECTION_PREFIX_LENGTH = 192;
+// allows for 4 full prefixes before hitting half of the max chunk size (approx.
+// 256 chars for 512 token chunks)
+const MAX_PREFIX_TOKENS = EMBEDDING_CONFIG.max_chunk_size / 8;
+// Limit on chars to avoid tokenizing too much text uselessly on documents with
+// large prefixes. The final truncating will rely on MAX_PREFIX_TOKENS so this
+// limit can be large and should be large to avoid underusing prexfixes
+const MAX_PREFIX_CHARS = MAX_PREFIX_TOKENS * 8;
 
 // The role of this function is to create a prefix from an arbitrary long string. The prefix
 // provided will not be augmented with `\n`, so it should include appropriate carriage return. If
-// the prefix is too long (>256 chars), it will be truncated. The remained will be returned as
+// the prefix is too long (> MAX_PREFIX_TOKENS), it will be truncated. The remained will be returned as
 // content of the resulting section.
-export function renderPrefixSection(
+export async function renderPrefixSection(
+  dataSourceConfig: DataSourceConfig,
   prefix: string | null
-): CoreAPIDataSourceDocumentSection {
+): Promise<CoreAPIDataSourceDocumentSection> {
   if (!prefix || !prefix.trim()) {
     return {
       prefix: null,
       content: null,
       sections: [],
     };
   }
-  if (prefix.length > MAX_SECTION_PREFIX_LENGTH) {
+  let finalPrefix = prefix;
```

Comment:
Let's trim()

Reply:
Actually, trimming here is problematic since callers have to add their carriage return for headings *after* calling renderPrefix (which they don't do now) -- and callers have to be aware of this function trimming which is not obvious (calling `renderPrefix(myprefix + "\n") seems natural and won't work, while this was the behaviour previously advocated by the comment

My preference would be to not trim here, let the callers trim if they think it is appropriate

Reply:
which is btw the case at least for notion

Reply:
Agreed 👍